tpu-client/Cargo.toml

```diff
@@ -21,9 +21,11 @@ solana-connection-cache = { workspace = true }
 solana-measure = { workspace = true }
 solana-metrics = { workspace = true }
 solana-pubsub-client = { workspace = true }
+solana-quic-client = { workspace = true }
 solana-rpc-client = { workspace = true }
 solana-rpc-client-api = { workspace = true }
 solana-sdk = { workspace = true }
+solana-udp-client = { workspace = true }
```

Comment:
I'd like to avoid adding both these dependencies to solana-tpu-client, so that downstream users don't have to import both by default if they only need one. I know you're trying to dedupe things (based on a reviewer request), but I think defining `QuicTpuClient` and `TpuClientWrapper` at the call sites or at least further upstream would be better.

Reply:
`QuicTpuClient` can be rearranged easily. `TpuClientWrapper` is a little harder. We could define it in `gossip_service` and then have `dos` import it along with `gossip/`. Although this requires `gossip` to import `solana-udp-client` and `solana-quic-client`. The other option I see is we put it back up in `solana-client` as before since `solana-client` is imported by `gossip_service` and `dos` anyway. But then this has the issue that we are now back to relying on `solana-client::tpu_client`...

Reply:
so I put `TpuClientWrapper` back in `solana_client` since `gossip` and `dos` are already importing `solana_client` anyway. I know it is not ideal. I also tried putting it in `gossip` but didn't like that we had to import `solana-udp-client` and `solana-quic-client` in gossip just for that wrapper. But if that is not a big deal, then lets add it in `gossip`. Couldn't find another spot further upstream either that made sense. Not sure the best path forward here.

Reply:
> but didn't like that we had to import solana-udp-client and solana-quic-client in gossip just for that wrapper

To be clear, the fact that `solana-gossip` imports `solana-client` means it is already importing `solana-udp-client` and `solana-quic-client` transitively. At some point it will actually be better to have it import one or both of those directly. But since it's still stuck on `solana_client::connection_cache::ConnectionCache`, this seems fine for now.

Reply:
oh shoot ya that's a good point. I should have realized that 🤦‍♂️. ok sounds good. Thank you!